web-ui/src/components/reports-section/CheckinReport.jsx

```diff
@@ -1,36 +1,188 @@
 } from '@mui/material';
 
+/**
+ * @typedef {Object} Checkin
+ * @property {string} id - The ID of the check-in.
+ * @property {boolean} completed - Indicates whether the check-in is completed.
+ * @property {Array} checkinDate - The date of the check-in.
+ * @property {string} pdlId - The ID of the PDL.
+ * @property {string} teamMemberId - The ID of the team member.
+ */
+
+/**
+ * @typedef {("Done" | "In Progress" | "Not Started")} CheckinStatus
+ * @typedef {("Not Yet Scheduled" | "Scheduled" | "Completed")} SchedulingStatus
+ */
+
+/** @type {CheckinStatus} */
+const steps = ['Not Started', 'In Progress', 'Done'];
+
+/** @type {SchedulingStatus} */
+const schedulingSteps = ['Not Yet Scheduled', 'Scheduled', 'Completed'];
+
+function HorizontalLinearStepper({ step = 0 }) {
+  const [activeStep, setActiveStep] = React.useState(step);
+  const [skipped, setSkipped] = React.useState(new Set());
+
+  const isStepOptional = step => step === -1;
+  const isStepSkipped = step => skipped.has(step);
+
+  return (
+    <Box sx={{ width: '100%', my: 1 }}>
+      <Stepper activeStep={activeStep}>
+        {steps.map((label, index) => {
+          const stepProps = {};
+          const labelProps = {};
+          if (isStepOptional(index)) {
+            labelProps.optional = (
+              <Typography variant="caption">Optional</Typography>
+            );
+          }
+          if (isStepSkipped(index)) {
+            stepProps.completed = false;
+          }
+          return (
+            <Step key={label} {...stepProps}>
+              <StepLabel {...labelProps}>{label}</StepLabel>
+            </Step>
+          );
+        })}
+      </Stepper>
+    </Box>
+  );
+}
+
 import './CheckinReport.css';
 
+const propTypes = {
+  closed: PropTypes.bool,
+  pdl: PropTypes.shape({
+    name: PropTypes.string,
+    id: PropTypes.string,
+    members: PropTypes.array,
+    workEmail: PropTypes.string,
+    title: PropTypes.string
+  }),
+  planned: PropTypes.bool
+};
+
 const CheckinsReport = ({ closed, pdl, planned }) => {
   const { state } = useContext(AppContext);
-  const { name, id, members, workEmail } = pdl;
+  const [expanded, setExpanded] = useState(true);
+  const [statusForPeriod, setStatusForPeriod] = useState(
+    /** @type CheckinStatus */ ('Not Started')
+  );
+
+  const { name, id, members, workEmail, title } = pdl;
+
+  const handleExpandClick = () => setExpanded(!expanded);
 
   const getCheckinDate = checkin => {
     if (!checkin || !checkin.checkInDate) return;
     const [year, month, day, hour, minute] = checkin.checkInDate;
     return new Date(year, month - 1, day, hour, minute, 0);
   };
 
+  /**
+   * Determine the status of the check-ins for a member.
+   * @param {Object} params - The parameters object.
+   * @param {Checkin[]} params.checkins - Checkins for a member.
+   * @returns {CheckinStatus} The status of check-ins.
+   */
+  const statusForPeriodByMember = ({ checkins = [] }) => {
+    const now = new Date();
```

Comment:
Micro-optimizing, but I would move line 121 to after line 124 so I `Date` object is only created if it will be needed.

Reply:
I like it. Took the optimization.